lib/livebook/settings.ex

```diff
@@ -269,4 +269,15 @@ defmodule Livebook.Settings do
   """
   @spec default_file_system_home() :: FileSystem.File.t()
   def default_file_system_home(), do: FileSystem.File.new(default_file_system())
+
+  def save_custom_protocol(protocol) do
+    storage().insert(:settings, "protocols", protocol: protocol)
```

Comment:
Should we store in "global" instead?  Also please add `@spec` and `@doc` as the other functions.